app/src/main/java/com/yacgroup/yacguide/AscendActivity.kt

```diff
@@ -45,63 +47,60 @@ import java.util.Calendar
 class AscendActivity : AppCompatActivity() {
 
     private lateinit var _db: DatabaseWrapper
-    private var _ascend: Ascend? = null
+    private lateinit var _ascend: Ascend
+    private var _existingAscend: Ascend? = null
     private var _route: Route? = null
-    private var _partnerIds: ArrayList<Int>? = null
-    private var _styleId: Int = 0
-    private var _year: Int = 0
-    private var _month: Int = 0
-    private var _day: Int = 0
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_ascend)
 
         _db = DatabaseWrapper(this)
 
-        _ascend = _db.getAscend(intent.getIntExtra(IntentConstants.ASCEND_KEY, DatabaseWrapper.INVALID_ID))
+        // The database query is null, if this activity is called from the route DescriptionActivity.
+        _existingAscend = _db.getAscend(intent.getIntExtra(IntentConstants.ASCEND_KEY,
+                DatabaseWrapper.INVALID_ID))
+        _ascend = _existingAscend ?: Ascend().apply { routeId = DatabaseWrapper.INVALID_ID }
+        // The intent constant ROUTE_KEY is only available, if this activity is called from
+        // the route DescriptionActivity.
         val routeId = intent.getIntExtra(IntentConstants.ROUTE_KEY, DatabaseWrapper.INVALID_ID)
-
         _route = _db.getRoute(routeId.takeUnless { it == DatabaseWrapper.INVALID_ID }
-                ?: _ascend?.routeId ?: DatabaseWrapper.INVALID_ID)
+                ?: _ascend.routeId)
         // Beware: _route may still be null (if the route of this ascend has been deleted meanwhile)
-        _partnerIds = intent.getIntegerArrayListExtra(IntentConstants.ASCEND_PARTNER_IDS)
-                ?: _ascend?.partnerIds ?: ArrayList()
 
-        findViewById<View>(R.id.notesEditText).onFocusChangeListener = View.OnFocusChangeListener { view, _ ->
-            val imm = view.context.getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
-            imm.hideSoftInputFromWindow(view.windowToken, 0)
+        findViewById<EditText>(R.id.notesEditText).let {
+            it.onFocusChangeListener = View.OnFocusChangeListener { view, _ ->
+                val imm = view.context.getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
+                imm.hideSoftInputFromWindow(view.windowToken, 0)
+            }
+            it.addTextChangedListener(object : TextWatcher {
```

Comment:
This is ok but a little bit like "mit Kanonen auf Spatzen". Why not saving all stuff to `_ascend` when this activity is paused for adding partners (callback `onPause()`)?

Reply:
Thanks for the hint. I just did not know about this feature.

Reply:
I like my solution more because otherwise we have two different locations where we have to save the text to the ascend object. First time in the possible `onPause` method and a second time in the `enter` method.

Reply:
ok for me. Maybe we can unify both in the future.